storage: TestReplicateRemovedNodeDisruptiveElection fails in prop-evaluated-kv

```
 make stress COCKROACH_PROPOSER_EVALUATED_KV=1 PKG=./pkg/storage TESTS=TestReplicateRemovedNodeDisruptiveElection
go list -tags '' -f \
	'go test -v  -tags '\'''\'' -ldflags '\'''\'' -i -c {{.ImportPath}} -o {{.Dir}}/stress.test && (cd {{.Dir}} && if [ -f stress.test ]; then stress -stderr -maxfails 1 ./stress.test -test.run '\''TestReplicateRemovedNodeDisruptiveElection'\'' -test.timeout 2m ; fi)' ./pkg/storage | \
	/usr/local/bin/bash
github.com/cockroachdb/cockroach/pkg/storage
github.com/cockroachdb/cockroach/pkg/storage_test
testmain
3 runs so far, 0 failures, over 5s
3 runs so far, 0 failures, over 10s
3 runs so far, 0 failures, over 15s
3 runs so far, 0 failures, over 20s
3 runs so far, 0 failures, over 25s
3 runs so far, 0 failures, over 30s
3 runs so far, 0 failures, over 35s
3 runs so far, 0 failures, over 40s
3 runs so far, 0 failures, over 45s

W161109 17:38:23.267644 1 storage/replica.go:102  running with experimental support for proposer-evaluated KV, see #10431
I161109 17:38:23.270852 1 rand.go:76  Random seed: -550417964603861076
I161109 17:38:23.273231 36 storage/engine/rocksdb.go:340  opening in memory rocksdb instance
I161109 17:38:23.274936 36 gossip/gossip.go:237  [n?] initial resolvers: []
W161109 17:38:23.274968 36 gossip/gossip.go:1055  [n?] no resolvers found; use --join to specify a connected node
I161109 17:38:23.275008 36 base/node_id.go:62  NodeID set to 1
I161109 17:38:23.278194 36 storage/store.go:1183  [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I161109 17:38:23.278285 36 gossip/gossip.go:280  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:54002" > attrs:<> locality:<>
I161109 17:38:23.279454 48 storage/replica_proposal.go:328  [s1,r1/1:/M{in-ax}] new range lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 900.000124ms following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0s [physicalTime=1970-01-01 00:00:00.000000123 +0000 UTC]
I161109 17:38:23.280028 36 storage/engine/rocksdb.go:340  opening in memory rocksdb instance
I161109 17:38:23.281083 36 gossip/gossip.go:237  [n?] initial resolvers: [127.0.0.1:54002]
W161109 17:38:23.281105 36 gossip/gossip.go:1057  [n?] no incoming or outgoing connections
I161109 17:38:23.281136 36 base/node_id.go:62  NodeID set to 2
I161109 17:38:23.283720 36 storage/store.go:1183  [n2,s2]: failed initial metrics computation: [n2,s2]: system config not yet available
I161109 17:38:23.283765 36 gossip/gossip.go:280  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:54009" > attrs:<> locality:<>
I161109 17:38:23.284224 122 gossip/client.go:125  [n2] started gossip client to 127.0.0.1:54002
I161109 17:38:23.284673 36 storage/engine/rocksdb.go:340  opening in memory rocksdb instance
I161109 17:38:23.285752 36 gossip/gossip.go:237  [n?] initial resolvers: [127.0.0.1:54002]
W161109 17:38:23.285767 36 gossip/gossip.go:1057  [n?] no incoming or outgoing connections
I161109 17:38:23.285795 36 base/node_id.go:62  NodeID set to 3
I161109 17:38:23.302873 36 storage/store.go:1183  [n3,s3]: failed initial metrics computation: [n3,s3]: system config not yet available
I161109 17:38:23.302937 36 gossip/gossip.go:280  [n3] NodeDescriptor set to node_id:3 address:<network_field:"tcp" address_field:"127.0.0.1:54019" > attrs:<> locality:<>
I161109 17:38:23.303606 195 gossip/client.go:125  [n3] started gossip client to 127.0.0.1:54002
I161109 17:38:23.303835 36 storage/engine/rocksdb.go:340  opening in memory rocksdb instance
I161109 17:38:23.305053 36 gossip/gossip.go:237  [n?] initial resolvers: [127.0.0.1:54002]
W161109 17:38:23.305079 36 gossip/gossip.go:1057  [n?] no incoming or outgoing connections
I161109 17:38:23.305154 36 base/node_id.go:62  NodeID set to 4
I161109 17:38:23.308837 36 storage/store.go:1183  [n4,s4]: failed initial metrics computation: [n4,s4]: system config not yet available
I161109 17:38:23.308880 36 gossip/gossip.go:280  [n4] NodeDescriptor set to node_id:4 address:<network_field:"tcp" address_field:"127.0.0.1:54031" > attrs:<> locality:<>
I161109 17:38:23.309557 257 gossip/client.go:125  [n4] started gossip client to 127.0.0.1:54002
I161109 17:38:23.311795 36 storage/replica_raftstorage.go:445  [s1,r1/1:/M{in-ax}] generated snapshot 6718ff57 at index 15 in 35.139µs.
I161109 17:38:23.312423 36 storage/store.go:3121  streamed snapshot: kv pairs: 34, log entries: 5
I161109 17:38:23.312716 276 storage/replica_raftstorage.go:589  [s2,r1/?:{-}] applying preemptive snapshot at index 15 (id=6718ff57, encoded size=16, 1 rocksdb batches, 5 log entries)
I161109 17:38:23.313128 276 storage/replica_raftstorage.go:592  [s2,r1/?:/M{in-ax}] applied preemptive snapshot in 0.000s
I161109 17:38:23.313628 36 storage/replica_command.go:3244  change replicas: read existing descriptor range_id:1 start_key:"" end_key:"\377\377" replicas:<node_id:1 store_id:1 replica_id:1 > next_replica_id:2
I161109 17:38:23.314626 241 storage/replica.go:2048  [s1,r1/1:/M{in-ax}] proposing ADD_REPLICA {NodeID:2 StoreID:2 ReplicaID:2}: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:2 StoreID:2 ReplicaID:2}]
I161109 17:38:23.315274 36 storage/replica_raftstorage.go:445  [s1,r1/1:/M{in-ax}] generated snapshot 2d8eb606 at index 17 in 32.041µs.
I161109 17:38:23.316710 36 storage/store.go:3121  streamed snapshot: kv pairs: 37, log entries: 7
I161109 17:38:23.316940 278 storage/replica_raftstorage.go:589  [s3,r1/?:{-}] applying preemptive snapshot at index 17 (id=2d8eb606, encoded size=16, 1 rocksdb batches, 7 log entries)
I161109 17:38:23.317387 278 storage/replica_raftstorage.go:592  [s3,r1/?:/M{in-ax}] applied preemptive snapshot in 0.000s
I161109 17:38:23.319075 36 storage/replica_command.go:3244  change replicas: read existing descriptor range_id:1 start_key:"" end_key:"\377\377" replicas:<node_id:1 store_id:1 replica_id:1 > replicas:<node_id:2 store_id:2 replica_id:2 > next_replica_id:3
I161109 17:38:23.320419 311 storage/raft_transport.go:432  raft transport stream to node 1 established
I161109 17:38:23.322259 302 storage/replica.go:2048  [s1,r1/1:/M{in-ax}] proposing ADD_REPLICA {NodeID:3 StoreID:3 ReplicaID:3}: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:2 StoreID:2 ReplicaID:2} {NodeID:3 StoreID:3 ReplicaID:3}]
I161109 17:38:23.323465 36 storage/replica_raftstorage.go:445  [s1,r1/1:/M{in-ax}] generated snapshot 2731dab1 at index 19 in 34.404µs.
I161109 17:38:23.324826 36 storage/store.go:3121  streamed snapshot: kv pairs: 40, log entries: 9
I161109 17:38:23.325109 325 storage/replica_raftstorage.go:589  [s4,r1/?:{-}] applying preemptive snapshot at index 19 (id=2731dab1, encoded size=16, 1 rocksdb batches, 9 log entries)
I161109 17:38:23.325678 325 storage/replica_raftstorage.go:592  [s4,r1/?:/M{in-ax}] applied preemptive snapshot in 0.001s
I161109 17:38:23.328516 36 storage/replica_command.go:3244  change replicas: read existing descriptor range_id:1 start_key:"" end_key:"\377\377" replicas:<node_id:1 store_id:1 replica_id:1 > replicas:<node_id:2 store_id:2 replica_id:2 > replicas:<node_id:3 store_id:3 replica_id:3 > next_replica_id:4
I161109 17:38:23.329659 269 storage/raft_transport.go:432  raft transport stream to node 1 established
I161109 17:38:23.331191 355 storage/replica.go:2048  [s1,r1/1:/M{in-ax}] proposing ADD_REPLICA {NodeID:4 StoreID:4 ReplicaID:4}: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:2 StoreID:2 ReplicaID:2} {NodeID:3 StoreID:3 ReplicaID:3} {NodeID:4 StoreID:4 ReplicaID:4}]
I161109 17:38:23.338931 36 storage/replica_command.go:3244  change replicas: read existing descriptor range_id:1 start_key:"" end_key:"\377\377" replicas:<node_id:1 store_id:1 replica_id:1 > replicas:<node_id:2 store_id:2 replica_id:2 > replicas:<node_id:3 store_id:3 replica_id:3 > replicas:<node_id:4 store_id:4 replica_id:4 > next_replica_id:5
I161109 17:38:23.340203 376 storage/raft_transport.go:432  raft transport stream to node 1 established
I161109 17:38:23.343657 382 storage/replica.go:2048  [s1,r1/1:/M{in-ax}] proposing REMOVE_REPLICA {NodeID:1 StoreID:1 ReplicaID:1}: [{NodeID:4 StoreID:4 ReplicaID:4} {NodeID:2 StoreID:2 ReplicaID:2} {NodeID:3 StoreID:3 ReplicaID:3}]
I161109 17:38:23.347427 341 storage/replica_proposal.go:631  [s1,r1/1:/M{in-ax}] unable to gossip first range; hasLease=false, err=range 1: replica {0 0 0} not lease holder; <nil> is
I161109 17:38:23.436128 330 storage/store.go:2978  [s1,r1/1:/M{in-ax}] added to replica GC queue (peer suggestion)
I161109 17:38:23.685804 409 storage/raft_transport.go:432  raft transport stream to node 3 established
I161109 17:38:23.686012 410 storage/raft_transport.go:432  raft transport stream to node 4 established
I161109 17:38:23.686472 418 storage/raft_transport.go:432  raft transport stream to node 2 established
I161109 17:38:23.686568 415 storage/raft_transport.go:432  raft transport stream to node 2 established
I161109 17:38:23.688657 111 storage/replica_proposal.go:328  [s2,r1/2:/M{in-ax}] new range lease replica {2 2 2} 1970-01-01 00:00:00.900000124 +0000 UTC 1.800000002s following replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 900.000124ms [physicalTime=1970-01-01 00:00:01.800000125 +0000 UTC]
I161109 17:38:23.688909 111 storage/replica_proposal.go:377  [s2,r1/2:/M{in-ax}] range [n2,s2,r1/2:/M{in-ax}]: transferring raft leadership to replica ID 4
I161109 17:38:23.689176 111 storage/replica_proposal.go:377  [s2,r1/2:/M{in-ax}] range [n2,s2,r1/2:/M{in-ax}]: transferring raft leadership to replica ID 3
I161109 17:38:23.689422 111 storage/replica_proposal.go:328  [s2,r1/2:/M{in-ax}] new range lease replica {2 2 2} 1970-01-01 00:00:00.900000124 +0000 UTC 1.800000002s following replica {3 3 3} 1970-01-01 00:00:00.900000124 +0000 UTC 1.800000002s [physicalTime=1970-01-01 00:00:01.800000125 +0000 UTC]
I161109 17:38:23.689435 179 storage/replica_proposal.go:328  [s3,r1/3:/M{in-ax}] new range lease replica {3 3 3} 1970-01-01 00:00:00.900000124 +0000 UTC 1.800000002s following replica {4 4 4} 1970-01-01 00:00:00.900000124 +0000 UTC 1.800000002s [physicalTime=1970-01-01 00:00:01.800000125 +0000 UTC]
I161109 17:38:23.690192 105 storage/replica_proposal.go:377  [s2,r1/2:/M{in-ax}] range [n2,s2,r1/2:/M{in-ax}]: transferring raft leadership to replica ID 4
I161109 17:38:23.691264 243 storage/replica_proposal.go:328  [s4,r1/4:/M{in-ax}] new range lease replica {4 4 4} 1970-01-01 00:00:00.900000124 +0000 UTC 1.800000002s following replica {2 2 2} 1970-01-01 00:00:00.900000124 +0000 UTC 1.800000002s [physicalTime=1970-01-01 00:00:01.800000125 +0000 UTC]
I161109 17:38:23.692436 394 storage/raft_transport.go:432  raft transport stream to node 4 established
I161109 17:38:23.692756 243 storage/replica_proposal.go:328  [s4,r1/4:/M{in-ax}] new range lease replica {4 4 4} 1970-01-01 00:00:00.900000124 +0000 UTC 1.800000002s following replica {2 2 2} 1970-01-01 00:00:00.900000124 +0000 UTC 1.800000002s [physicalTime=1970-01-01 00:00:01.800000125 +0000 UTC]
I161109 17:38:23.693688 401 storage/raft_transport.go:432  raft transport stream to node 3 established
E161109 17:38:23.696791 36 storage/client_test.go:981  engine 0: missing key "a"
I161109 17:38:35.909998 243 storage/replica.go:2095  [s4,r1/4:/M{in-ax}] not quiescing: 1 pending commands
I161109 17:38:47.371834 46 gossip/gossip.go:1071  [n1] node has connected to cluster via gossip
I161109 17:38:55.712604 220 gossip/gossip.go:1071  [n4] node has connected to cluster via gossip
I161109 17:38:59.391697 102 gossip/gossip.go:1071  [n2] node has connected to cluster via gossip
I161109 17:38:59.874817 143 gossip/gossip.go:1071  [n3] node has connected to cluster via gossip
I161109 17:39:08.858583 5699 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
I161109 17:39:08.858647 5699 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
I161109 17:39:08.858744 5699 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
I161109 17:39:08.858806 5699 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
I161109 17:39:08.858864 5699 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
I161109 17:39:08.859120 212 http2_server.go:276  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:54031->127.0.0.1:54032: use of closed network connection
I161109 17:39:08.859172 164 http2_server.go:276  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:54019->127.0.0.1:54024: use of closed network connection
I161109 17:39:08.859176 94 http2_server.go:276  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:54009->127.0.0.1:54012: use of closed network connection
W161109 17:39:08.859229 394 storage/raft_transport.go:437  raft transport stream to node 4 failed: rpc error: code = 13 desc = transport is closing
W161109 17:39:08.859244 418 storage/raft_transport.go:437  raft transport stream to node 2 failed: rpc error: code = 13 desc = transport is closing
I161109 17:39:08.859310 38 http2_server.go:276  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:54002->127.0.0.1:54006: use of closed network connection
I161109 17:39:08.859355 5699 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
I161109 17:39:08.859517 5699 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
I161109 17:39:08.859606 5699 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
I161109 17:39:08.859697 5699 util/stop/stopper.go:396  stop has been called, stopping or quiescing all running tasks
--- FAIL: TestReplicateRemovedNodeDisruptiveElection (45.59s)
	testing.go:117: storage/client_raft_test.go:2271, condition failed to evaluate within 45s: leaseReplica 4 does not match leadReplica 3
FAIL


ERROR: exit status 1

4 runs completed, 1 failures, over 46s
FAIL
make: *** [stress] Error 1
```

Since I'm adding prop-evaluated-kv to CI, I'm going to disable this test in #10576.

--- a/pkg/storage/client_raft_test.go
+++ b/pkg/storage/client_raft_test.go
@@ -2383,10 +2383,6 @@ func (errorChannelTestHandler) HandleSnapshot(
 func TestReplicateRemovedNodeDisruptiveElection(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 
-	if storage.ProposerEvaluatedKVEnabled() {
-		t.Skip("#10577")
-	}
-
 	mtc := &multiTestContext{}
 	defer mtc.Stop()
 	mtc.Start(t, 4)

